public/model/epi-data-store.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Stores epidemiological/case data.
+ *
+ * NOTE! This is a super hyper rough sketch. Data is *not* fetched from the
+ * backend, but just hardcoded for now. We still need to decide on a reasonable
+ * schema to account for temporal and spatial spottiness and inconsistencies.
+ */
+
+var _ = require('lodash');
+var P = require('pjs').P;
+var Q = require('q');
+
+var DateUtil = require('./date-util.js');
+
+var utc_date = function(year, month, day) {
+  // Note: months are zero-indexed.
+  return new Date(Date.UTC(year, month - 1, day));
+};
+
+// TODO(jetpack): screaming for unit tests.
+
+var EpiDataStore = P({
+  init: function(on_update) {
+    this.on_update = on_update;
+
+    // `data_by_date_interval_and_region` format is an array of objects that looks
+    // like this:
+    // {start_time: <Date>, end_time: <Date>, region_case_data: <region case data>}
+    //
+    // The array is sorted by `start_time`. The times are all at midnight UTC.
+    // The time interval is half-open: [start_time, end_time). So, with
+    // start_time = 2016-01-01 and end_time = 2016-01-08, the interval contains
+    // data for 2016-01-01 up to and including 2016-01-07.
+    //
+    // Region case data is a mapping from region code -> condition ->
+    // culmulative case count. For example:
+    // {br1: {dengue: 120, malaria: 110, zika: 100},
+    //  br2: {dengue: 220, malaria: 210},
+    //  br3: {malaria: 310, zika: 300}}
+    this.data_by_date_interval_and_region = [];
+    this.most_recent_start_time = null;
+
+    this.initial_load_promise = Q.delay(10).then((function() {
+      // 4589 is Iguape, 4611 is Itanhaém, and 4877 is São Paulo.
+      this.data_by_date_interval_and_region = [
```

Comment:
this unit-y data is probably better for tests. otherwise, please make sure the user knows if the real data failed to load -- that could be confusing.